src/Type/IntersectionType.php

```diff
@@ -842,7 +842,7 @@ public function getEnumCases(): array
 		foreach ($this->types as $type) {
 			$oneType = [];
 			foreach ($type->getEnumCases() as $enumCase) {
-				$oneType[md5($enumCase->describe(VerbosityLevel::typeOnly()))] = $enumCase;
+				$oneType[$enumCase->getClassName() . '::' . $enumCase->getEnumCaseName()] = $enumCase;
```

Comment:
I think since we only handle enum case objects here, which are always final, we should be able to take this shortcut